### PR TITLE
Implementing (parts of) the proposal from #676

### PR DESCRIPTION
**Pull request by ibabushkin**
Targeting #676:
1. Changed representation of register pairs for the `m68k` architecture
2. Aligned members of `M68K_OP_TYPE` enum and the tagged union in `cs_m68k_op` struct.
3. fixed some small compiler warnings in test suite (irrelevant to main goal).

### CODE REVIEW

File: arch/M68K/M68KDisassembler.c [hunk 1]
```diff
@@ -2789,7 +2792,8 @@ static void d68020_mull(m68k_info *info)
 
 	op1->address_mode = M68K_AM_NONE;
 	op1->type = M68K_OP_REG_PAIR;
-	op1->register_bits = (reg_0 << 4) | reg_1;
+	op1->reg_pair.reg_0 = reg_0;
+	op1->reg_pair.reg_1 = reg_1;
```
Comment (emoon) [inline arch/M68K/M68KDisassembler.c]: Make sure to update this code also https://github.com/ibabushkin/capstone/blob/next/arch/M68K/M68KInstPrinter.c#L114 after these changes.

**Comment by ibabushkin** [comment]
Updated code to fit the requirements discussed previously in #676 and above.

File: arch/M68K/M68KDisassembler.c (hunk 1 above)
Comment (ibabushkin) [inline arch/M68K/M68KDisassembler.c]: Done.

Comment (emoon) [inline arch/M68K/M68KDisassembler.c]: :+1:

**Comment by ibabushkin** [comment]
ping @aquynh: so how is the status of this? :)

**Comment by aquynh** [comment]
this looks good, but bindings will break after this change.

can you please fix the Python binding for me?

**Comment by ibabushkin** [comment]
sure, I'll look into them. Interesting how I always forget that part ;)

**Comment by ibabushkin** [comment]
Alright, the python bindings should be fixed now.

**Comment by ibabushkin** [comment]
The python bindings should be ok now.

**Comment by aquynh** [comment]
merged, thanks!

however, when running `tests/test_m68k` or `bindings/python/test_m68k.py`, there are few output lines like below.

```
0x101e:	fmove.s	#1000.000000, fp0
	op_count: 2
	groups_count: 0
		operands[1].type: REG = fp0
```

why the first operand is missing here?

**Comment by ibabushkin** [comment]
That's interesting. I doubt the changes introduced are the reason for this, but I will investigate.

**Comment by emoon** [comment]
This is likely due to the change of type in your changes which make this code: https://github.com/aquynh/capstone/blob/next/tests/test_m68k.c#L87 not to work anymore.

**Comment by ibabushkin** [comment]
Changing the type of operands shouldn't affect their amount in an instruction, or did I misunderstand the issue?

**Comment by emoon** [comment]
This is because before all IMM values where set to ```M68K_OP_IMM``` (even if they are FPU) when printing the actual value it was checked which type of IMM value it was. Now with your changes you instead of being IMM it would be ```FPU_SINGLE/FPU_DOUBLE``` which the printing code does/didn't handle.

**Comment by ibabushkin** [comment]
Yeah, I got so far. Further cleaning and python tests are coming soon.

**Comment by emoon** [comment]
👍

**Pull request merged**